Some quality of life additions

wtlgo opened the pull request:
- Add ability to preserve the input history between runs
- Add ability to input numeric values directly
- Add ability to set the input before running

Commit:
Add ability to input numeric values directly

Commit:
Add ability to reuse the previous input

Commit:
Add ability to set input before running

Commit:
Fix typo

arkark commented on src/components/BrainfuckEngine.vue:
```diff
@@ -106,19 +106,43 @@
     <form @submit.prevent="input" style="margin: 12px 0px">
       <b-field>
-        <b-input v-model="inputText" :disabled="!isRunning"> </b-input>
+        <b-input v-model="inputText" :disabled="isRunning && inPause">
```
```suggestion
        <b-input v-model="inputText" :disabled="!isRunning">
```

I'd like to this change to be reverted because input texts before running are deleted in the `run` method:
https://github.com/arkark/brainfuck-online-simulator/blob/42f1ee76164f13e8e356c0bc329b89902783802a/src/components/BrainfuckEngine.vue#L286

arkark commented on src/components/BrainfuckEngine.vue:
```diff
@@ -106,19 +106,43 @@
+        </b-input>
         <div class="control">
-          <button class="button is-dark" :disabled="!isRunning">Input</button>
+          <button class="button is-dark" :disabled="isRunning && inPause">
```
```suggestion
          <button class="button is-dark" :disabled="!isRunning">
```

arkark commented on src/components/BrainfuckEngine.vue:
```diff
@@ -106,19 +106,43 @@
+      <b-field>
+        <b-input
+          v-model.number="inputCode"
+          :disabled="isRunning && inPause"
```
```suggestion
          :disabled="!isRunning"
```

arkark commented on src/components/BrainfuckEngine.vue:
```diff
@@ -106,19 +106,43 @@
+        >
+        </b-input>
+        <div class="control">
+          <button class="button is-dark" :disabled="isRunning && inPause">
```
```suggestion
          <button class="button is-dark" :disabled="!isRunning">
```

arkark commented on src/components/BrainfuckEngine.vue:
```diff
@@ -249,33 +298,56 @@ export default {
+    },
+    input: function () {
+      this.addText(this.inputText);
+      this.inputHistory = [
+        ...this.inputHistory,
+        { type: "text", val: this.inputText },
+      ];
```
```suggestion
      this.inputHistory.push({
        type: "text",
        val: this.inputText,
      });
```
In this case, the spread syntax inefficient and redundant.

arkark commented on src/components/BrainfuckEngine.vue:
```diff
@@ -249,33 +298,56 @@ export default {
     },
+    inputCharCode: function () {
+      this.addCharCode(this.inputCode);
+      this.inputHistory = [
+        ...this.inputHistory,
+        { type: "code", val: this.inputCode },
+      ];
```
```suggestion
      this.inputHistory.push({
        type: "code",
        val: this.inputCode,
      });
```

arkark commented on src/components/BrainfuckEngine.vue:
```diff
@@ -106,19 +106,43 @@
+      </b-field>
+    </form>
+
+    <b-checkbox v-model="repeatHistory"> Repeat History </b-checkbox>
```
- Add `:disabled="isRunning` to this `<b-checkbox>`, because changing this checkbox has no effect while running.
- Move this `<b-checkbox>` to the bottom of `Input History` for UI/UX.

arkark commented on src/components/BrainfuckEngine.vue:
```diff
@@ -106,19 +106,43 @@
+        </b-input>
+        <div class="control">
+          <button class="button is-dark" :disabled="isRunning && inPause">
+            Input Value
```
```suggestion
            Input Char Code
```
I think `Char Code` is more obvious than `Value`.

arkark reviewed (changes requested):
Thank you for your contribution.

The new features are nice. But please fix what I pointed out.

wtlgo committed:
Update src/components/BrainfuckEngine.vue

Co-authored-by: Takeshi Kaneko <16665605+arkark@users.noreply.github.com>

wtlgo committed:
Update src/components/BrainfuckEngine.vue

Co-authored-by: Takeshi Kaneko <16665605+arkark@users.noreply.github.com>

wtlgo committed:
Update src/components/BrainfuckEngine.vue

Co-authored-by: Takeshi Kaneko <16665605+arkark@users.noreply.github.com>

Commit:
Move this checkbox to the bottom

Commit:
Change input disabling

wtlgo commented:
Hello! Thank you for your feedback. I fixed everything except one thing. 
Based on my personal experience the ability to manipulate the input before running the program makes using the simulator _much_ easier, that's why I believe this feature should stay. So instead of rolling it back, I changed the way it works a little. Please, check it out, I hope you'll like it more.

Commit:
Make isInputDisabled consistent

arkark reviewed (approved):
You may have a point there. https://github.com/arkark/brainfuck-online-simulator/pull/4/commits/7158d14895a77f066a5b76f982f854a0ad679659 is good. 

Thank you!
